src/main/webapp/app/shared/statistics-graph/statistics-average-score-graph.component.ts

```diff
@@ -40,20 +48,102 @@ export class StatisticsAverageScoreGraphComponent implements OnInit {
-        this.ngxData = this.exerciseAverageScores
-            .slice(this.currentPeriod, 10 + this.currentPeriod)
-            .map((exercise, index) => ({ name: this.barChartLabels[index], value: exercise.averageScore }));
+        this.ngxData = this.exerciseAverageScores.slice(this.currentPeriod, 10 + this.currentPeriod).map(
+            (exercise, index) =>
+                ({
+                    name: this.barChartLabels[index],
+                    value: exercise.averageScore,
+                    exerciseType: exercise.exerciseType,
+                    exerciseId: exercise.exerciseId,
+                } as ExerciseStatisticsEntry),
+        );
+        this.ngxColor.domain = this.ngxData.map((exercise) => this.determineColor(exercise.value));
     }
 
     // handles arrow clicks and updates the exercises which are shown, forward is boolean since it is either forward or backward
     public switchTimeSpan(forward: boolean): void {
         this.currentPeriod += forward ? 1 : -1;
         this.initializeChart();
     }
+
+    /**
+     * Determines the color of the bar given the score
+     * @param score that is represented by the bar
+     * @returns string rgba representation of the color
+     * @private
+     */
+    private determineColor(score: number): string {
+        return score >= 90 ? GraphColors.GREEN : score <= 25 ? GraphColors.RED : GraphColors.DARK_BLUE;
+    }
+
+    /**
+     * Handles the click event on one of the bars and navigates to the corresponding exercise statistics page
+     * @param event the event that is passed by the framework
+     */
+    onSelect(event: any): void {
+        const dataEntry = this.determineChartEntry(event.name, event.value);
+
+        // a workaround in order to prevent false navigation. If more than one exercise is matching, no routing is done
+        if (dataEntry) {
+            const route = ['course-management', this.courseId, '', dataEntry.exerciseId, 'exercise-statistics'];
+            let type = dataEntry.exerciseType.toLowerCase();
+            if (type === ExerciseType.QUIZ) {
+                route[4] = 'quiz-point-statistic';
+            } else if (type === 'file_upload') {
+                type = 'file-upload';
+            }
+            route[2] = type + '-exercises';
+            this.router.navigate(route);
+        }
+    }
+
+    /**
+     * Determines the entry in the chart for a mouse event
+     * @param name name of the exercise
+     * @param value average score of the exercise
+     * @private
+     */
+    private determineChartEntry(name: string, value: number): ExerciseStatisticsEntry | undefined {
+        let counter = 0;
+        let result;
+        /**
+         * this is not a good solution as neither the name of the exercise nor its average score is necessarily unique.
+         * But as the exercise type is not included in the emitted event, I do not see a better solution.
+         */
```

Comment:
```suggestion
        /*
         * The emitted event only contains the name and the average score of the exercise. Using those values to determine the chart entry
         * is not an ideal solution as this pair is not necessarily unique.
         * In practice they most likely are unique, though. Not being able to find the entry in this edge case therefore has negligible impact.
         */
```
I think not handling this special edge case is okay.